src/nix-store/nix-store.cc

```diff
@@ -364,6 +369,32 @@ static void opQuery(Strings opFlags, Strings opArgs)
             }
             break;
 
+        case qUncached: {
+            if (opArgs.empty()) throw UsageError("no paths passed");
+
+            ThreadPool pool(fileTransferSettings.httpConnections);
+
+            std::function<void(StorePath)> queryPath;
+            queryPath = [&](const StorePath & req) {
+                SubstitutablePathInfos infos;
+                store->querySubstitutablePathInfos({{req, std::nullopt}}, infos);
+
+                if (infos.empty()) {
+                    cout << store->printStorePath(req) << '\n';
+                }
+            };
+
+            for (auto & i : opArgs) {
+                auto path = store->followLinksToStorePath(i);
+
+                pool.enqueue(std::bind(queryPath, path));
+            }
```

Comment:
Does anything speak against leaving out the `std::function` and `std::bind` by writing it like this: ?

```cpp
            auto queryPathThunk = [&](const StorePath & req) {
                return [&]{
                    SubstitutablePathInfos infos;
                    store->querySubstitutablePathInfos({{req, std::nullopt}}, infos);

                    if (infos.empty()) {
                        cout << store->printStorePath(req) << '\n';
                    }
                };
            };

            for (auto & i : opArgs) {
                pool.enqueue(queryPathThunk(store->followLinksToStorePath(i)));
            }
```

This would remove some typing information that is both unneeded and fixed otherwise.